src/api/app/models/event/base.rb

```diff
@@ -246,7 +246,7 @@ def watchers
       project = ::Project.find_by_name(payload['project'])
       return [] if project.blank?
 
-      project.watched_projects.map(&:user).concat(project.watched_items.map(&:user)).uniq
+      project.watched_items.map(&:user)
```

Comment:
This code generates N+1 queries, see below an example from my local development environment:

```
  Project Load (0.3ms)  SELECT `projects`.* FROM `projects` WHERE `projects`.`id` != 0 AND `projects`.`name` = 'home:Admin' LIMIT 1
  WatchedItem Load (0.3ms)  SELECT `watched_items`.* FROM `watched_items` WHERE `watched_items`.`watchable_id` = 3 AND `watched_items`.`watchable_type` = 'Project'
  User Load (0.3ms)  SELECT `users`.* FROM `users` WHERE `users`.`id` = 1 LIMIT 1
  User Load (0.4ms)  SELECT `users`.* FROM `users` WHERE `users`.`id` = 2 LIMIT 1
  User Load (0.3ms)  SELECT `users`.* FROM `users` WHERE `users`.`id` = 3 LIMIT 1
  User Load (0.3ms)  SELECT `users`.* FROM `users` WHERE `users`.`id` = 4 LIMIT 1
  User Load (0.3ms)  SELECT `users`.* FROM `users` WHERE `users`.`id` = 5 LIMIT 1
  User Load (0.3ms)  SELECT `users`.* FROM `users` WHERE `users`.`id` = 6 LIMIT 1
  User Load (0.3ms)  SELECT `users`.* FROM `users` WHERE `users`.`id` = 7 LIMIT 1
  User Load (0.3ms)  SELECT `users`.* FROM `users` WHERE `users`.`id` = 8 LIMIT 1
  User Load (0.3ms)  SELECT `users`.* FROM `users` WHERE `users`.`id` = 9 LIMIT 1
```

To avoid this, we can include the `user` association for the `watched_items`:

```suggestion
      project.watched_items.includes(:user).map(&:user)
```

This results in two queries hitting the database:

```
  WatchedItem Load (0.3ms)  SELECT `watched_items`.* FROM `watched_items` WHERE `watched_items`.`watchable_id` = 3 AND `watched_items`.`watchable_type` = 'Project'
  User Load (0.4ms)  SELECT `users`.* FROM `users` WHERE `users`.`id` IN (1, 2, 3, 4, 5, 6, 7, 8, 9)
```